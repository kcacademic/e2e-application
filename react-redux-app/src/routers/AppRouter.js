import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Header from '../components/Header';
import MyCloud from '../components/MyCloud';
import DashBoard from '../components/DashBoard';
import NotFound from '../components/NotFound';

const AppRouter = () => (
    <BrowserRouter>
        <div className='container'>
            <Header />
            <Switch>
                <Route path="/" component={DashBoard} exact={true} />
                <Route component={NotFound} />
            </Switch>
        </div>
    </BrowserRouter>
);

export default AppRouter;