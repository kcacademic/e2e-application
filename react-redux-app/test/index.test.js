import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import AppRouter from '../src/routers/AppRouter';
import getAppStore from '../src/store/store';

test('Dashboard component with TagCloud rendered.', () => {

    const store = getAppStore();
    const component = renderer.create(
        <Provider store={store}>
            <AppRouter />
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});