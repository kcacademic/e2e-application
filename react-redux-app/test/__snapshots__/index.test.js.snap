// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component with TagCloud rendered. 1`] = `
<div
  className="container"
>
  <header
    style={
      Object {
        "flex": 1,
        "textAlign": "center",
      }
    }
  >
    <h2>
      Sample Tag Cloud
    </h2>
    <h4>
      Twitter Word Frequency Analysis
    </h4>
  </header>
  <div
    className="container__list"
  >
    <div>
      <div>
        <div
          className="app-outer"
        >
          <div
            className="app-inner"
          >
            <div
              className="tag-cloud"
              style={
                Object {
                  "color": [Function],
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <ul />
      </div>
    </div>
  </div>
</div>
`;
